engines/unknown-game-dialog.cpp

```diff
@@ -0,0 +1,144 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "common/translation.h"
+#include "common/str-array.h"
+#include "gui/gui-manager.h"
+#include "gui/message.h"
+#include "gui/ThemeEval.h"
+#include "gui/widgets/popup.h"
+#include "engines/unknown-game-dialog.h"
+#include "backends/platform/sdl/sdl.h"
+
+namespace GUI {
+
+enum {
+	kCopyToClipboard = 'cpcl',
+	kOpenBugtrackerURL = 'ourl',
+	kClose = 'clse'
+};
+
+UnknownGameDialog::UnknownGameDialog(const Common::String &reportData, const Common::String &reportTranslated, const Common::String &bugtrackerAffectedEngine)
+	: Dialog(30, 20, 260, 124) {
+
+	_reportData = reportData;
+	_reportTranslated = reportTranslated;
+	_bugtrackerAffectedEngine = bugtrackerAffectedEngine;
+
+	//Check if we have clipboard functionality and expand the reportTranslated message if needed...
+	if (g_system->hasFeature(OSystem::kFeatureClipboardSupport)) {
+		_reportTranslated += _("Use the button below to copy the required game information into your clipboard. \n");
+	}
+
+	//Check if we have support for opening URLs and expand the reportTranslated message if needed...
+	if (g_system->hasFeature(OSystem::kFeatureOpenUrl)) {
+		_reportTranslated += _("You can also directly report your game to the Bug Tracker!");
+	}
+
+	const int screenW = g_system->getOverlayWidth();
+	const int screenH = g_system->getOverlayHeight();
+
+	int buttonWidth = g_gui.xmlEval()->getVar("Globals.Button.Width", 0);
+	int buttonHeight = g_gui.xmlEval()->getVar("Globals.Button.Height", 0);
+
+	//Calculate the size the dialog needs
+	Common::Array<Common::String> lines;
+	int maxlineWidth = g_gui.getFont().wordWrapText(_reportTranslated, screenW - 2 * 20, lines);
+
+	_w = MAX(MAX(maxlineWidth, 0), (2 * buttonWidth) + 10) + 20;
+	int lineCount = lines.size() + 1;
+
+	_h =  3 * kLineHeight;
+	_h += lineCount * kLineHeight;
+
+	// Center the dialog
+	_x = (screenW - _w) / 2;
+	_y = (screenH - _h) / 2;

```

Comment:
Centering the dialog needs to happen in the `reflowLayout` method so the dialog stays centered when resizing the window.